platform/admin/core/admin-core-appservice/src/main/java/org/codice/ddf/admin/application/service/impl/ApplicationImpl.java

```diff
@@ -95,6 +99,19 @@ public ApplicationImpl(Repository repo) {
             name = mainFeature.getName();
             version = mainFeature.getVersion();
             description = mainFeature.getDescription();
+        } else if (autoInstallFeatures.size() == 0) {
+            // If the app's features file does not contain any auto install features, set
+            // mainFeature to the first feature found in the file.
```

Comment:
If we really wanted to be compatible with regular feature files, shouldn't we be attempting to calculate the root feature of a features file as a dependency graph? I'm wondering about the potential side-effects of making this assumption about the first feature, although perhaps the container does enough dependency traversal that it is safe.